doc/conf.py

```diff
@@ -1655,6 +1657,7 @@ def setup(app):
     """Set up the Sphinx app."""
     app.connect("autodoc-process-docstring", append_attr_meth_examples)
     # High prio, will happen before SG
+    app.connect("builder-inited", generate_credit_rst, priority=10)
```

Comment:
FYI @drammock I ended up changing the structure a bit. I realized that having a `credit.rst` in our code history that gets updated with thousands of badges is less than ideal -- it'll change a lot and bloat things I think. So instead now I have it create a `doc/code_credit.inc` on the fly (it's very fast) at the start of the doc build so that we don't have to keep it in our history. In theory this could instead be a directive, but I think it's actually nicer as a `.inc` because you can inspect it after it's generated, don't have to deal with docutils and nested parsing at all, etc.